src/stream/src/task/stream_manager.rs

```diff
@@ -284,21 +289,35 @@ impl LocalStreamManager {
         Ok(())
     }
 
-    pub fn drop_actor(&self, actors: &[ActorId]) -> StreamResult<()> {
+    pub fn drop_actor(
+        &self,
+        source_mgr: &dyn TableSourceManager,
+        actors: &[ActorId],
+    ) -> StreamResult<()> {
         let mut core = self.core.lock();
         for id in actors {
             core.drop_actor(*id);
+            if let Some(table_id) = core.actor_tables.remove(id) {
+                source_mgr.try_drop_source(&table_id);
```

Comment:
There are some other ways where there's no change on the stream manager. 😄

- Store an `Option`al reference of source manager in `SourceDesc`. In `impl Drop`, remove the entry in the table source manager.
- On every `insert_source`, scan all entries in the map and `drain_filter` the ones that cannot be upgraded.

Do you think this will be better? 🤔